precompiles/erc20/types.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package erc20
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// ParseTransferArgs parses the arguments from the transfer method and returns
+// the destination address (to) and amount.
+func ParseTransferArgs(args []interface{}) (
+	to common.Address, amount *big.Int, err error,
+) {
+	if len(args) != 2 {
+		return common.Address{}, nil, fmt.Errorf("invalid number of arguments; expected 2; got: %d", len(args))
+	}
+
+	to, ok := args[0].(common.Address)
+	if !ok {
+		return common.Address{}, nil, fmt.Errorf("invalid to address: %v", args[0])
+	}
+
+	amount, ok = args[1].(*big.Int)
+	if !ok {
+		return common.Address{}, nil, fmt.Errorf("invalid amount")
+	}
+
+	return to, amount, nil
+}
+
+// ParseTransferFromArgs parses the arguments from the transferFrom method and returns
+// the sender address (from), destination address (to) and amount.
+func ParseTransferFromArgs(args []interface{}) (
+	from common.Address, to common.Address, amount *big.Int, err error,
+) {
+	if len(args) != 3 {
+		return common.Address{}, common.Address{}, nil, fmt.Errorf("invalid number of arguments; expected 3; got: %d", len(args))
+	}
+
+	from, ok := args[0].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, fmt.Errorf("invalid from address: %v", args[0])
+	}
+
+	to, ok = args[1].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, fmt.Errorf("invalid to address: %v", args[0])
+	}
+
+	amount, ok = args[2].(*big.Int)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, fmt.Errorf("invalid amount")
```

Comment:
would prefer to also print the arg here: 
```suggestion
		return common.Address{}, common.Address{}, nil, fmt.Errorf("invalid amount: %v", args[2])
```